### PR TITLE
kafka-matrix: convert to mzworkflows.py

### PR DESCRIPTION
**Pull request by benesch**
Touches #9809.

### Motivation

   * This PR refactors existing code.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

**Comment by benesch** [comment]
Nightly build: https://buildkite.com/materialize/nightlies/builds/553

**Review by philip-stoev — APPROVED** [review]
This looks good, but I was under the impression that re-defining services is a big no-no in docker/docker-compose?

Some time ago, in a review for another mzworkflows.py, you said you prefer that all the participating services are listed in the original `services` list , so I coded that thing (the upgrade test?) to prepare a single comprehensive `services` list at the start of the test.

**Comment by benesch** [comment]
Thanks very much for reminding me of my own rule. I've adjusted this PR so that it is only possible to _overwrite_ an existing service definition. That alleviates my concern, because `./mzcompose down -v` will happily bring overridden services down, and `./mzcompose up` will restore them to their default configurations if they've been overwritten by a workflow. Sorry for not thinking of this middle ground before! I've adjusted the upgrade test to work like this for consistency's sake.

File: test/kafka-matrix/mzworkflows.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+CONFLUENT_PLATFORM_VERSIONS = [
+    "4.0.0",
+    "5.0.0",
+    "6.0.2",
+    "6.1.2",
+    "6.2.0",
+    "latest",
+]
+
+services = [
```
Comment (guswynn) [inline test/kafka-matrix/mzworkflows.py:28]: is this a special ident that mzworkflow expects to import?

Comment (benesch) [reply test/kafka-matrix/mzworkflows.py:28]: Yup! There's documentation on all this here! https://github.com/MaterializeInc/materialize/blob/main/doc/developer/mzworkflows.md

**Pull request merged**